### PR TITLE
Unit Testing - Finished my part of tests for repositories

### PR DESCRIPTION
**Pull request by leonidbogin**

### CODE REVIEW

**Comment by leonidbogin** [comment]
All my tests are working
image

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DIMS_Core.Tests.DataAccessLayer.Repositories
+{
+    [TestFixture]
+    public class TaskTrackRepositoryTest : RepositoryTestBase
+    {
+        private TaskTrackRepository query;
```
Comment (SashaDrapun) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs:15]: I think this field should be readonly

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+        public void GetAll_ActualCount()
+        {
+            // Arrange
+            int expected = Context.TaskTrack.Count();
+
+            // Act
+            var actual = query.GetAll();
+
+            // Assert
+            Assert.That(expected, Is.EqualTo(actual.Count()));
```
Comment (SashaDrapun) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs:33]: I consider code comments unnecessary

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            // Assert
+            Assert.IsNull(actual);
+        }
+
+        [Test]
+        public async Task Update_Id1_NewTrackNote()
+        {
+            // Arrange
+            int updateId = 1;
+            string expected = "New track note";
```
Comment (SashaDrapun) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs:119]: Strings can be made constants for optimization

**Review by SashaDrapun — COMMENTED** [review]
Hi, I have noted the points that seemed to me wrong

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+            // Arrange
+            int getId = 5;
+
+            // Act
+            var actual = await query.GetByIdAsync(getId);
+
+            // Assert
+            Assert.IsNull(actual);
+        }
+
```
Comment (SashaDrapun) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs:62]: User Assert.That(actual, Is.Null)

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs (hunk 1 above)
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs:15]: It is not query

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+            await query.DeleteAsync(deleteId);
+            Context.SaveChanges();
+            var actual = await query.GetByIdAsync(deleteId);
+
+            // Assert
+            Assert.IsNull(actual);
+        }
+
+        [OneTimeTearDown]
+        public void CleanupQuery()
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs:216]: It is not query

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+            await query.CreateAsync(expected);
+            Context.SaveChanges();
+            var actual = await query.GetByIdAsync(newId);
+
+            // Assert
+            Assert.That(expected, Is.EqualTo(actual));
+        }
+
+        [Test]
+        public void CreateAsync_Null_NotThrow()
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs:93]: Async??

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/UserProfileRepositoryTests.cs [hunk 7]
```diff
@@ -7,107 +7,219 @@
...
 
-            //Assert
-            Assert.AreEqual(newUserProfile, result);
+            // Assert
+            Assert.That(expected, Is.EqualTo(actual));
         }
 
         [Test]
-        public async Task ShouldUpdate()
+        public void CreateAsync_Null_NotThrow()
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/UserProfileRepositoryTests.cs:97]: Async??

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/UserTaskRepositoryTest.cs [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+            await query.CreateAsync(expected);
+            Context.SaveChanges();
+            var actual = await query.GetByIdAsync(newId);
+
+            // Assert
+            Assert.That(expected, Is.EqualTo(actual));
+        }
+
+        [Test]
+        public void CreateAsync_Null_NotThrow()
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/UserTaskRepositoryTest.cs:95]: Async??

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi man!
I did some comments. Look them through and fix.

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs (hunk 2 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs]: I removed arrange, act, assert comments

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs (hunk 3 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs]: I changed string to const string

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs (hunk 4 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs]: I changed Assert.IsNull(actual) to Assert.That(actual, Is.Null).

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs (hunk 1 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs:15]: I changed query to repository

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs (hunk 5 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs:180]: I changed query to repository

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs (hunk 6 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs:84]: CreateAsync is method's name (<Method>_<Action>_<Result>)

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/UserProfileRepositoryTests.cs (hunk 7 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/UserProfileRepositoryTests.cs:88]: CreateAsync is method's name (Method_Action_Result)

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/UserTaskRepositoryTest.cs (hunk 8 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/UserTaskRepositoryTest.cs:86]: CreateAsync is method's name (Method_Action_Result)

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs (hunk 6 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/TaskTrackRepositoryTest.cs:84]: Сhanged the name of the test method to the correct:
public void CreateAsync_WithNullEntryShouldReturn_NotThrow()

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/UserProfileRepositoryTests.cs (hunk 7 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/UserProfileRepositoryTests.cs:88]: Сhanged the name of the test method to the correct:
public void CreateAsync_WithNullEntryShouldReturn_NotThrow()

File: DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/UserTaskRepositoryTest.cs (hunk 8 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.Tests/DataAccessLayer/Repositories/UserTaskRepositoryTest.cs:86]: Сhanged the name of the test method to the correct:
public void CreateAsync_WithNullEntryShouldReturn_NotThrow()

**Pull request merged**